discord/bot.py

```diff
@@ -191,270 +198,583 @@ def get_application_command(
+            guild_ids: List[int] = None
+    ) -> None:
         """|coro|
-        Registers all commands that have been added through :meth:`.add_application_command`.
-        This method cleans up all commands over the API and should sync them with the internal cache of commands.
-        This will only be rolled out to Discord if :meth:`.http.get_global_commands` has certain keys that differ from :data:`.pending_application_commands`
-        By default, this coroutine is called inside the :func:`.on_connect`
-        event. If you choose to override the :func:`.on_connect` event, then
-        you should invoke this coroutine as well.
+
+        Registers a command. If the command has guild_ids set, or if the guild_ids parameter is passed, the command will
+        be registered as a guild command for those guilds.
+
+        Parameters
+        ----------
+        command: :class:`~.ApplicationCommand`
+            The command to register.
+        force: :class:`bool`
+            Whether to force the command to be registered. If this is set to False, the command will only be registered
+            if it seems to already be registered and up to date with our internal cache. Defaults to True.
+        guild_ids: :class:`list`
+            A list of guild ids to register the command for. If this is not set, the command's
+            :attr:`~.ApplicationCommand.guild_ids` attribute will be used.
+
+        Returns
+        -------
+        :class:`~.ApplicationCommand`
+            The command that was registered
+        """
+        # TODO: Write this
+        return
+
+    async def register_commands(
+            self,
+            commands: Optional[List[ApplicationCommand]] = None,
+            guild_id: Optional[int] = None,
+            force: bool = False
+    ):
+        """|coro|
+
+        Register a list of commands.
+
         .. versionadded:: 2.0
+
+        Parameters
+        ----------
+        commands: Optional[List[:class:`~.ApplicationCommand`]]
+            A list of commands to register. If this is not set (None), then all commands will be registered.
+        guild_id: Optional[int]
+            If this is set, the commands will be registered as a guild command for the respective guild. If it is not
+            set, the commands will be registered according to their :attr:`~.ApplicationCommand.guild_ids` attribute.
+        force: :class:`bool`
+            Registers the commands regardless of the state of the command on discord, this can take up more API calls
+            but is sometimes a more foolproof method of registering commands. This also sometimes causes minor bugs
+            where the command can temporarily appear as an invalid command on the user's side. Defaults to False.
         """
-        commands_to_bulk = []
+        if commands is None:
+            commands = self.pending_application_commands
 
-        needs_bulk = False
+        commands = copy.deepcopy(commands)
```

Comment:
When using deepcopy with a list of commands it raises `TypeError: cannot pickle 'generator' object`.

Reply:
sounds like you passed a generator, not a list